internal/errorlint/errorlint.go

```diff
@@ -0,0 +1,191 @@
+// Copyright 2023 Outreach Corporation. All Rights Reserved.
+
+// Description: See package comment for this one file package.
+
+// Package errorlint contains the necessary logic for the error/log/trace linter. The error
+// errors.New, errors.WithMessage, errors.WithMessagef, trace.StartCall, trace.StartSpan, log.Warn,
+// log.Error, log.Info, errors.Wrap, errors.Wrapf, fmt.Errorf, errors.Errorf static messages for
+// capitalization. It also checks messages ending in punctuation ". : ! and \n"
+package errorlint
+
+import (
+	"go/ast"
+	"go/token"
+	"strconv"
+	"unicode"
+	"unicode/utf8"
+
+	"github.com/getoutreach/lintroller/internal/common"
+	"github.com/getoutreach/lintroller/internal/reporter"
+	"golang.org/x/tools/go/analysis"
+)
+
+// name defines the name for the errorlint linter.
+const name = "errorlint"
+
+// doc defines the help text for the error linter.
+const doc = `Ensures that each static message in error/trace/log/ is lowercase.
+
+A valid example is the following:
+
+	errors.New("org not found in launchdarkly rule")
+An invalid example is the following:
+
+	errors.New("Org not found in LAUNCHDARKLY rule")
+An invalid example is the following:
+
+	errors.New("Org not found in launchdarkly rule:")`
```

Comment:
Make sure you document the punctuation rule in the text. I'd also re-use Google's example:
```suggestion
const doc = `Ensures that each error message starts with a lower-case letter and does not end in puctuation.

// Bad:
err := fmt.Errorf("Something bad happened.")

// Good:
err := fmt.Errorf("something bad happened")
```

Reply:
added